lib/transbank/sdk/onepay/models/transaction.rb

```diff
@@ -1,73 +1,87 @@
-                                                             external_unique_number,
-                                                             options)
-          response = Utils::NetHelper.post(transaction_create_path, JSON.parse(create_request.jsonify))
+
+          options = complete_options(options)
+          create_request = create_transaction(shopping_cart: shopping_cart,
+                                              channel: channel,
+                                              external_unique_number: external_unique_number,
+                                              options: options)
+          puts "CREATE REQUEST IS:\n #{create_request.to_h}"
+          response = http_post(transaction_create_path, create_request.to_h)
 
           validate_create_response!(response)
           transaction_create_response = TransactionCreateResponse.new response
-          validate_signature!(transaction_create_response, options)
+          signature_is_valid = transaction_create_response.validate_signature(options.fetch(:shared_secret))
+          unless signature_is_valid
+            raise Errors::SignatureError, "The response's signature is not valid."
+          end
           transaction_create_response
         end
 
-        # TODO: Document
-        def commit(occ, external_unique_number, options = nil)
-          options = Utils::RequestBuilder.complete_options(options)
-          commit_request = Utils::RequestBuilder.commit_transaction(occ, external_unique_number, options)
-          response = Utils::NetHelper.post(transaction_commit_path, JSON.parse(commit_request.jsonify))
+        # Commit a [Transaction]. It is MANDATORY for this to be done, and you have
+        # 30 seconds to do so, otherwise the [Transaction] is automatically REVERSED
+        # @param occ [String] Merchant purchase order
+        # @param external_unique_number [String] a unique value (per Merchant, not global) that is used to identify a Transaction
+        # @param options[Hash, nil] an optional Hash with configuration overrides
+        # @return [TransactionCommitResponse] The response to your commit request.
+        # @raise [TransactionCommitError] if response is nil or responseCode of the response is not 'OK'
+        def commit(occ:, external_unique_number:, options: nil)
+          options = complete_options(options)
+          commit_request = commit_transaction(occ: occ,
+                                              external_unique_number: external_unique_number,
+                                              options: options)
+          response = http_post(transaction_commit_path, commit_request.to_h)
           validate_commit_response!(response)
           transaction_commit_response = TransactionCommitResponse.new(response)
-          validate_signature!(transaction_commit_response, options)
+          signature_is_valid = transaction_commit_response.validate_signature(options.fetch(:shared_secret))
```

Comment:
I'd rename that method from `validate_signature` to `valid_signature?`

Reply:
:+1: